### PR TITLE
Use transport abstraction and transport-abstracted references

### PR DESCRIPTION
**Pull request by mtrmac**
This demonstrates https://github.com/containers/image/pull/33 .

### CODE REVIEW

File: cmd/skopeo/utils.go [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
-const (
-	// atomicPrefix is the URL-like schema prefix used for Atomic registry image references.
-	atomicPrefix = "atomic:"
-	// dockerPrefix is the URL-like schema prefix used for Docker image references.
-	dockerPrefix = "docker://"
-	// directoryPrefix is the URL-like schema prefix used for local directories (for debugging)
-	directoryPrefix = "dir:"
-)
+// parseImageName converts URL-like image name to an types.ImageReference.
+func parseImageName(imgName string) (types.ImageReference, error) {
```
Comment (mtrmac) [inline cmd/skopeo/utils.go]: Actually this function feels like it should be provided by `containers/image`.

Comment (runcom) [inline cmd/skopeo/utils.go]: +1

Comment (mtrmac) [inline cmd/skopeo/utils.go]: Done, creating `containers/image/transports.ParseImageName`. The `transports` package is, sadly, probably necessary.

Comment (runcom) [inline cmd/skopeo/utils.go]: I like `transports`

**Comment by runcom** [comment]
:+1:

**Pull request merged**